homeassistant/components/esphome/__init__.py

```diff
@@ -493,14 +495,23 @@ def __init__(self, entry_id: str, component_key: str, key: int):
             async_dispatcher_connect(
                 self.hass,
                 DISPATCHER_ON_DEVICE_UPDATE.format(**kwargs),
-                self.async_schedule_update_ha_state,
+                self._on_device_update,
             )
         )
 
-    async def _on_update(self) -> None:
+    async def _on_state_update(self) -> None:
```

Comment:
Why is that a coroutine? You can mark the function as `@callback` and the dispatcher execute it inside loop.

Reply:
Mostly because the Camera entity type needs to override this function and executes coroutine code.

Reply:
I agree that in normal project it does not have an impact without have a benchmark. For Core with process many events per seconds, it have one ;)